lib/inc/hocon/config_mergeable.hpp

```diff
@@ -6,6 +6,7 @@
 namespace hocon {
 
     class LIBCPP_HOCON_EXPORT config_mergeable {
+        friend class config_value;
```

Comment:
Ok, that maybe makes sense. I'm going to play with this a bit, we shouldn't need to `friend class` when calling a method of a class the method inherits from.